File: src/basic_training/Day17_5.java
package basic_training;

import java.util.Arrays;

public class Day17_5 {
	public String[] solution(String my_string) {
		String[] answer = my_string.split(" "); // 공백 제거해야함
		return Arrays.stream(answer).filter(item -> !item.equals("")).toArray(String[]::new);
	}
}

// equals("") == isEmpty()

// 참고할만한 번외 문제풀이 _ trim(), split()
// return my_string.trim().split("\\s+");
// 문자열의 양끝 공백을 제거 및 하나이상의 공백문자를 기준으로 나눈 배열반환

// trim() : 문자열 양 끝의 공백을 제거
// \s 공백(정규표현식)
// \s+ 하나이상의 공백문자

File: src/basic_training/Day17_5.js
function solution(my_string) {
	return my_string.split(" ").filter(item => item !== "");
}

// 동일한 결과
// return my_string.split(" ").filter(item => item);
// item 이 true 인 값만 반환
// "" 공백은 false 이므로 값에서 제외되므로
// 자바스크립트에서 'falsy' 값은 false, 0, ""(빈 문자열), null, undefined, NaN 


// 번외문제풀이 _ 공백 정규표현식
// return my_string.trim().split(/\s+/);

// /는 정규 표현식 패턴의 시작과 끝을 나타내며, \s+는 하나 이상의 공백 문자

